rdflib/plugins/serializers/turtle.py

```diff
@@ -284,6 +284,26 @@ def getQName(self, uri, gen_prefix=True):
 
             if pfx is not None:
                 parts = (pfx, uri, "")
+
+            # Does the uri contain chars that need to be escaped?
+            elif len(self.namespaces) > 0:
+                for prefix, local in [
+                    (_prefix, uri.split(_namespace)[1])
+                    for _prefix, _namespace in self.namespaces.items()
```

Comment:
Why is this using `self.namespaces` instead of `self.store.namespaces` ? I'm guessing if you do that it will work in all cases.

Even so, while I think that may fix the parameterizes tests I added, I still don't think it is ideal to have logic to compute a CURIE inside the recovery handler for the function that should be computing the CURIE (i.e. `compute_qname`).

I feel it is making this function more complicated than it has to be and already I'm a bit unsure of the complexity. To me for example it seems wrong that this function is doing `self.addNamespace(prefix, namespace)` - and I'm also unclear on why the serializer has `addNamespace` and a list of namespaces.

Reply:
If I change this to use `self.store.namespaces` it works fine with the parameterized tests I added in https://github.com/gjhiggins/rdflib/compare/fix-issue-948...aucampia:rdflib:iwana-20220721T2317-fix_turtle_serialization